tt_metal/hw/firmware/src/erisc.cc

```diff
@@ -28,13 +28,50 @@ uint32_t wIndex __attribute__((used));
 
 tt_l1_ptr mailboxes_t * const mailboxes = (tt_l1_ptr mailboxes_t *)(eth_l1_mem::address_map::ERISC_MEM_MAILBOX_BASE);
 
+uint8_t noc_index = 0;  // TODO: remove hardcoding
 uint8_t my_x[NUM_NOCS] __attribute__((used));
 uint8_t my_y[NUM_NOCS] __attribute__((used));
 
 uint32_t noc_reads_num_issued[NUM_NOCS] __attribute__((used));
 uint32_t noc_nonposted_writes_num_issued[NUM_NOCS] __attribute__((used));
 uint32_t noc_nonposted_writes_acked[NUM_NOCS] __attribute__((used));
 
+FORCE_INLINE
+void multicore_eth_cb_wait_front(db_cb_config_t *eth_db_cb_config, int32_t num_pages) {
+    DEBUG_STATUS('C', 'R', 'B', 'W');
+
+    uint16_t pages_received;
+    do {
+        pages_received = uint16_t(eth_db_cb_config->recv - eth_db_cb_config->ack);
+    } while (pages_received < num_pages);
+    DEBUG_STATUS('C', 'R', 'B', 'D');
+}
+
+FORCE_INLINE
+void multicore_eth_cb_pop_front(
+    db_cb_config_t *eth_db_cb_config,
+    const db_cb_config_t *remote_db_cb_config,
+    uint64_t producer_noc_encoding,
+    uint32_t num_pages) {
+    eth_db_cb_config->ack += num_pages;
+
+    uint32_t pages_ack_addr = (uint32_t)(&(remote_db_cb_config->ack));
+    noc_semaphore_set_remote((uint32_t)(&(eth_db_cb_config->ack)), producer_noc_encoding | pages_ack_addr);
+}
+
+FORCE_INLINE
+uint64_t get_dispatch_core_noc_encoding() {
+    uint32_t dispatch_noc_x = (uint32_t)((routing_info->dispatch_core_xy >> 16) & 0xFFFF);
+    uint32_t dispatch_noc_y = (uint32_t)(routing_info->dispatch_core_xy & 0xFFFF);
```

Comment:
I left one 16 byte chunk for metadata: 
```
    volatile uint32_t routing_enabled;
    volatile uint32_t routing_mode;
    volatile uint32_t connected_chip_id;
    volatile uint32_t dispatch_core_xy;
    ```
  And I didn't want to use another 16 bytes lol. I'll add more space to this, because I have a feeling we'll need more info here later on anyway